File: 2017/01/main.py
f = open("input.txt", "r")
s = f.read().strip()
# s += s[0]
ans = 0
# for i in range(len(s) - 1):
# 	if s[i] == s[i+1]:
# 		ans += int(s[i])
dist = len(s) // 2
for i in range(len(s)):
	idx = (i + dist) % len(s)
	if s[i] == s[idx]:
		ans += int(s[i])
		
print(ans)

File: 2017/01/main.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
  string s;
  cin >> s;
  s += s[0];
  cout << s << endl;
  int ans = 0;
  for (int i = 0; i < s.size() - 1; i++) {
    if (s[i] == s[i + 1]) {
      ans += s[i] - '0';
    }
  }
  cout << ans << endl;
}
